integrations/clickup/clickup/client.py

```diff
@@ -0,0 +1,254 @@
+from typing import Any, AsyncGenerator, List, Dict
+import httpx
+from loguru import logger
+from port_ocean.context import event
+from port_ocean.context.ocean import ocean
+from port_ocean.utils import http_async_client
+
+PAGE_SIZE = 50
+
+class ClickUpClient:
+    def __init__(self, clickup_url: str, clickup_api_key: str) -> None:
+        """
+        Initializes an instance of the ClickUpClient class.
+
+        Args:
+            clickup_url (str): The base URL for the ClickUp API.
+            clickup_api_key (str): The API key for authenticating requests to the ClickUp API.
+        """
```

Comment:
Which this is great, we want to keep the code base clean so you can remove the docs string and only add inline comment where necessary. We believe that function names need to be clear